src/sagemaker_tensorflow_container/training.py

```diff
@@ -159,6 +160,33 @@ def train(env):
                                   runner=runner_type)
 
 
+def _log_model_missing_warning(model_dir):
+    pb_file_exists = False
+    file_exists = False
+    for dirpath, dirnames, filenames in os.walk(model_dir):
+        if filenames:
+            file_exists = True
+        for f in filenames:
+            if 'saved_model.pb' in f or 'saved_model.pbtxt' in f:
+                pb_file_exists = True
+                path, direct_parent_dir = os.path.split(dirpath)
+                if not str.isdigit(direct_parent_dir):
+                    logger.warn('Your model will NOT be servable with SageMaker TensorFlow Serving containers.'
+                                'The SavedModel bundle is under directory \"{}\", not a numeric name.'
+                                .format(direct_parent_dir))
+
+    if not file_exists:
+        logger.warn('No model artifact is saved under path {}.'
```

Comment:
Shouldn't we toss an exception here?

Are there any use cases of users not utilizing /opt/ml/model for packaging their models to S3?

Reply:
They could have saved models or checkpoints to s3 during training. saving it to '/opt/ml/model' just to make sure ease will upload it at end of training.

Reply:
Can you provide any links or references to users expecting to have had their training artifacts in /opt/ml/model, however they didn't know?